core/src/main/java/org/infinispan/distribution/ch/impl/SingleSegmentKeyPartitioner.java

```diff
@@ -0,0 +1,26 @@
+package org.infinispan.distribution.ch.impl;
+
+import org.infinispan.distribution.ch.KeyPartitioner;
+
+/**
+ * KeyPartitioner that always returns 0 for a given segment. This can be useful when segments are not in use, such
+ * as local or invalidation caches.
+ * @author wburns
+ * @since 9.3
+ */
+public class SingleSegmentKeyPartitioner implements KeyPartitioner {
+   private SingleSegmentKeyPartitioner() { }
+
+   public static SingleSegmentKeyPartitioner getInstance() {
+      return SingletonHelper.INSTANCE;
+   }
+
+   @Override
+   public int getSegment(Object key) {
+      return 0;
+   }
+
+   private static class SingletonHelper {
+      private static final SingleSegmentKeyPartitioner INSTANCE = new SingleSegmentKeyPartitioner();
```

Comment:
Why hide it in another class?

Reply:
This way it is only initialized if we need it.

Reply:
Obviously low (and will be retained on the heap until class is unloaded), but why even have to create it if we don't have to :D

Reply:
As always, it's a balance act... a separate class will also use some heap and even more metaspace.

Reply:
Yeah for the initialized part, true.